packages/sit-onyx/src/components/OnyxSlider/OnyxSlider.vue

```diff
@@ -0,0 +1,725 @@
+<script setup lang="ts">
+import { createSlider } from "@sit-onyx/headless";
+import { iconMinusSmall, iconPlusSmall } from "@sit-onyx/icons";
+import { computed, toRefs } from "vue";
+import { useDensity } from "../../composables/density.js";
+import { getFormMessages, useCustomValidity } from "../../composables/useCustomValidity.js";
+import { useErrorClass } from "../../composables/useErrorClass.js";
+import {
+  SKELETON_INJECTED_SYMBOL,
+  useSkeletonContext,
+} from "../../composables/useSkeletonState.js";
+import { useVModel } from "../../composables/useVModel.js";
+import { useRootAttrs } from "../../utils/attrs.js";
+import { FORM_INJECTED_SYMBOL, useFormContext } from "../OnyxForm/OnyxForm.core.js";
+import OnyxFormElement from "../OnyxFormElement/OnyxFormElement.vue";
+import OnyxIconButton from "../OnyxIconButton/OnyxIconButton.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import OnyxTooltip from "../OnyxTooltip/OnyxTooltip.vue";
+import type { OnyxSliderProps } from "./types.js";
+
+const clamp = (value: number, min: number, max: number) => Math.min(Math.max(value, min), max);
+
+/**
+ * Adjusting the position for marks with proper edge offset to prevent overflow because of rounding.
+ * For marks at 0% and 100%, applies a 0.25rem offset to keep them within bounds.
+ */
+const adjustMarkPosition = (percentage: number): string => {
+  if (percentage <= 0) {
+    return "calc(0.25rem)";
+  }
+  if (percentage >= 100) {
+    return "calc(100% - 0.25rem)";
+  }
+  return `${percentage}%`;
+};
+
+const props = withDefaults(defineProps<OnyxSliderProps>(), {
+  min: 0,
+  max: 100,
+  step: 1,
+  shiftStep: 10,
+  marks: false,
+  modelValue: (props) => [props.min ?? 0] as number[],
+  orientation: "horizontal",
+  trackMode: "default",
+  tooltipDisplay: "auto",
+  disabled: FORM_INJECTED_SYMBOL,
+  showError: FORM_INJECTED_SYMBOL,
+  skeleton: SKELETON_INJECTED_SYMBOL,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the validity state of the slider changes.
+   */
+  validityChange: [validity: ValidityState];
+  /**
+   * Emitted when the slider changes
+   */
+  "update:modelValue": [value: number[]];
+  /**
+   * Emitted when the slider is committed (e.g. on blur, or on pressing enter)
+   */
+  "update:committed": [value: number[]];
+}>();
+
+const modelValue = useVModel({
+  props,
+  emit,
+  key: "modelValue",
+  default: () => [props.min ?? 0] as number[],
+});
+
+const customError = computed(() => props.customError);
+const { vCustomValidity, errorMessages } = useCustomValidity({ props, emit, customError });
+const messages = computed(() => getFormMessages(props.message));
+
+defineOptions({ inheritAttrs: false });
+
+const { rootAttrs, restAttrs } = useRootAttrs();
```

Comment:
```suggestion
```

Since there is no single logical component that the slider is wrapping, I don't think we need to split up the attributes here and can just use's Vue's defaults :)

Unlike the OnyxInput or OnyxButton for example, there it is clear that the rootAttrs should be forwarded to the native HTMl input/button